### DISCUSSION TITLE
compatibility with fzf-tab?

### DISCUSSION THREAD

**Discussion by fujianyawa** [marlonrichert/zsh-autocomplete#500] [Q&A] [answered]
Hello, thanks for the amazing plugin!
But how can I make  [fzf-tab](https://github.com/Aloxaf/fzf-tab) works alongside zsh-autocomplete?

Here's my .zshrc config:
```sh
source $ZSH/custom/plugins/zsh-autocomplete/zsh-autocomplete.plugin.zsh

if [[ -r "${XDG_CACHE_HOME:-$HOME/.cache}/p10k-instant-prompt-${(%):-%n}.zsh" ]]; then
  source "${XDG_CACHE_HOME:-$HOME/.cache}/p10k-instant-prompt-${(%):-%n}.zsh"
fi

export "PATH=${PATH}:${HOME}/.local/bin"
export ZSH="$HOME/.oh-my-zsh"

ZSH_THEME="powerlevel10k/powerlevel10k"
ZSH_TMUX_AUTOSTART="true"

plugins=(
  alias-tips
  extract
  fd
  git
  ripgrep
  thefuck
  tmux
  # web search
  zsh-autopair
  zsh-autocomplete
  zsh-autosuggestions
  zsh-completions
  zsh-syntax-highlighting
  # load this last to make the ** keybind work
  fzf
  # fzf-tab
)

fpath+=${ZSH_CUSTOM:-${ZSH:-~/.oh-my-zsh}/custom}/plugins/zsh-completions/src
source $ZSH/oh-my-zsh.sh

source /usr/share/zsh/site-functions/fzf
zstyle ':autocomplete:tab:*' fzf-completion yes
zstyle ':autocomplete:*' min-input 1

# my-fzf-tab() {
#   functions[compadd]=$functions[-ftb-compadd]
#   zle fzf-tab-complete
# }
# zle -N fzf-tab-complete
# bindkey "^I" fzf-tab-complete

[[ ! -f ~/.p10k.zsh ]] || source ~/.p10k.zsh
# import custom aliases and function
[ -f $HOME/.zsh_aliasrc ] && source $HOME/.zsh_aliasrc
[ -f $HOME/.profile ] && source $HOME/.profile

pokemon-colorscripts -r --no-title; tput setaf 4 && tput bold; fortune
typeset -g POWERLEVEL9K_INSTANT_PROMPT=quiet
```
Uncommenting this
`source $ZSH/custom/plugins/zsh-autocomplete/zsh-autocomplete.plugin.zsh` and  `plugins (#zsh-autocomplete ...)`
Makes the fzf-tab to work, but I want to use zsh-autocomplete too.

And also the author of the fzf-tab suggest [in this thread](https://github.com/Aloxaf/fzf-tab/issues/198#issuecomment-1288358158) to either choose which one of plugins to use, Though It's nice to use both of the plugins.

**Comment by marlonrichert** [comment]
`zstyle ':autocomplete:tab:*' fzf-completion yes` lets you use Zsh Autocomplete with [Fzf's official shell extensions](https://github.com/junegunn/fzf/tree/master/shell). It does not provide compatibility with Fzf Tab.

I'm curious: How would you even expect this to work in practice, from the user's point of view?

**Reply by fujianyawa** [reply]
> `zstyle ':autocomplete:tab:*' fzf-completion yes` lets you use Zsh Autocomplete with [Fzf's official shell extensions](https://github.com/junegunn/fzf/tree/master/shell). It does not provide compatibility with Fzf Tab.

Ah yes, that was supposed to be for the fzf completion (copied the snippet from [here](https://github.com/marlonrichert/zsh-autocomplete/discussions/202#discussioncomment-451705)). I just want `**` key bind to work, and now I've recently discovered fzf-tab and I really wanted to use that plugin for interactive fuzzily search alongside zsh-autocompletion, But I don't know what to do to make the fzf-tab works, I mean if it is possible that would be nice if not then I'll to stick on using this autocomplete

> I'm curious: How would you even expect this to work in practice, from the user's point of view?

Uhm, sorry? I didn't quite get your question very well (sorry english is not my native language, so sorry if there's a grammatical error in my response)

**Reply by marlonrichert** [reply]
> > I'm curious: How would you even expect this to work in practice, from the user's point of view?
>
> Uhm, sorry? I didn't quite get your question very well

When would you expect Zsh Autocomplete to be used and when Fzf Tab?

**Reply by fujianyawa** [reply]
> When would you expect Zsh Autocomplete to be used and when Fzf Tab?

~~Oh, I see I'll use Fzf tab if there's a lot a suggested result or a really long word I can fuzzily filter it and plus using it interactively is what i like I'm quite fond to fzf.  And for the zsh autocomplete with the ahead completion it is cool, but for me the caveats is when selecting a suggestion navigating via the arrow keys it was bad experience for me, if there's a lot of suggested output navigating it down one by one would take a while and like I say it was a bad experience because if I navigate down  it would automatically type the results for me but what if that's not what I'm looking for now I will try press backspace many times to erase the autocompleted output that i wasn't expected too ( I mean sure i can use alt+backspace to quickly clear the word) but it's such a hassle for and do type again If I couldn't find what I'm searching for. I think if the suggessted result filling up the screen searching fuzzily to look for the result is a nice option~~

[TL;DR](https://github.com/marlonrichert/zsh-autocomplete/issues/501#issuecomment-1292909686)

**Reply by marlonrichert** [reply]
Ah, I see. That should definitely be possible, but I’ll have to look into it a bit first to figure out exactly how to make it work. I’ll get back to you later.

**Reply by fujianyawa** [reply]
> Ah, I see. That should definitely be possible, but I’ll have to look into it a bit first to figure out exactly how to make it work. I’ll get back to you later.

Woah, cool... nice and Thank you very much!

**Reply by marlonrichert** [reply]
I made an issue for adding support: #501.

**Reply by matru** [reply]
I could not reply to #501 since it's locked. But I have an issue with fzf-tab compatbility. @marlonrichert you mentioned that everything should work out of the box, and if correct, tabbing should switch to fzf-tab menu while arrow keys to autocomplete. However this bugs out for me, if I restart shell it works fine until I missclick tab when there isn't any command typed in, then it's not possible to trigger fzf-tab with tab and autocomplete takes the lead, but in a buggy state, where pressing tab doesn't do anything, the listed options just blink, so I need to force another shell restart since now both things are rendered useless.

**Reply by rodhash** [reply]
> `zstyle ':autocomplete:tab:*' fzf-completion yes` lets you use Zsh Autocomplete with [Fzf's official shell extensions](https://github.com/junegunn/fzf/tree/master/shell). It does not provide compatibility with Fzf Tab.
> 

Is this still working? 

After enabling this I get the completions the standard way instead of with FzF:

<img width="289" alt="image" src="https://github.com/user-attachments/assets/72c30f61-e6ac-4bd6-a453-68d30f2210bc" />

**Reply by ArmandAgopian** [reply]
I am also unable to get this working anymore.

**Reply by marlonrichert** [reply]
I'm not able to get `fzf-tab` working at all, not even in a clean environment:

```zsh
% brew install fzf
# ...
% exec zsh
% cd $( mktemp -d )
% git clone git@github.com:Aloxaf/fzf-tab.git
# ...
% > .zshrc <<EOF
setopt interactivecomments transientrprompt
PS1='%# '
PS2=
RPS2='%^'
autoload compinit
compinit
source fzf-tab/fzf-tab.zsh
EOF
% env -i HOME=$PWD PATH=$PATH TERM=$TERM ${TERMINFO:+TERMINFO=$TERMINFO} zsh -d
% fzf  # Test that fzf works.
# It does.
% file # Press Tab here.
% file fzf-tab/  # I get normal completion, not fzf.
```

**Reply by marlonrichert** [reply]
I filed a bug report with `fzf-tab`: https://github.com/Aloxaf/fzf-tab/issues/532

**Reply by ArmandAgopian** [reply]
Interesting, it works for me in a clean environment. Did you verify the plugin is downloaded to the correct location?

**Comment by norsemangrey** [comment]
This is not working for me either.

**Comment by marlonrichert** [comment, marked as answer]
If you want another plugin to override Autocomplete's keybindings, then simply source that plugin _after_ you source Autocomplete.

However, I'm not able to say anything about the particular case of `fzf-tab`, because [I cannot get that plugin to work at all](https://github.com/marlonrichert/zsh-autocomplete/discussions/500#discussioncomment-13650236).
